### ISSUE TITLE
yolov4 low mAP when yolov4-tiny has very high accuracy on the same data

### ISSUE THREAD

**Issue by a-rich** [AlexeyAB/darknet#6689]
I'm training models on a custom dataset composed of images of people taken from an overhead camera. This is a single class model. As such, I've made a few minor changes to the tiny-yolov4.cfg and yolov4.cfg found in the repo...namely changing the number of filters prior to the prediction heads to be 18 instead of 255, classes from 80 to 1, and calculating custom anchors.

Anchors for the tiny model were calculated like this:
`./darknet detector calc_anchors ~/models/overhead/cfgs/2020-09-04/yolo.data -num_of_clusters 6 -width 384 -height 384`

Anchors for the full sized model were calculated like this:
`./darknet detector calc_anchors ~/models/overhead/cfgs/2020-09-16_full/yolo.data -num_of_clusters 9 -width 608 -height 608`

The training set has about 500,000 images in it (~31K original images augmented with flips, rotations, noise, hue, saturation, contrast, gamma, brightness, random erasing, cropping, etc.). These the labels for this dataset have been extensively audited...quality of the labels aside, the fact that the tiny model is vastly outperforming the full sized model tells me there's probably something wrong with the cfg for the full sized model, but I cannot figure out what it is.

I train with these shell functions:
```
burnin() {
    /home/lrm/repos/darknet/darknet detector train ${1}yolo.data ${1}yolov4_burnin.cfg -map -dont_show -gpus 0 -mjpeg_port 8090 > /home/lrm/repos/darknet/log.txt
    mv /home/lrm/repos/darknet/log.txt ${1}log_burnin.txt
    mv ${PWD}/chart.png ${1}chart_burnin.png
}

train() {
    /home/lrm/repos/darknet/darknet detector train ${1}yolo.data ${1}yolov4.cfg ${2} -map -dont_show -gpus 0,1 -mjpeg_port 8090 > /home/lrm/repos/darknet/log.txt
    mv /home/lrm/repos/darknet/log.txt ${1}log.txt
    mv ${PWD}/chart.png ${1}chart.png
}
```

So training the tiny model training is run like this:
`train ~/models/overhead/cfgs/2020-09-04/  ~/backup/overhead/2020-09-04/yolov4_burnin_1000.weights`

and the full sized model training is run like this:
`train ~/models/overhead/cfgs/2020-09-16_full/  ~/backup/overhead/2020-09-16_full/yolov4_burnin_1000.weights`

More truthfully, these shell functions and other operations are scripted:
```
#!/bin/bash -i
. ~/.bashrc
burnin ~/models/$1/cfgs/$2/ && \
train ~/models/$1/cfgs/$2/ ~/backup/$1/$2/yolov4_burnin_1000.weights && \
...
```

so the process of training (at least as far as run commands go) has very little room for error:
`./hit_that_switch.sh overhead 2020-09-04`

Full sized YOLOv4 model gets 35-45% mAP:
![yolov4_chart](https://user-images.githubusercontent.com/4131505/93652759-0d988c80-f9cb-11ea-9eaf-178d2c0f5501.png)

Tiny YOLOv4 model gets greater than 90% mAP on the same dataset:
![tiny-yolov4_chart](https://user-images.githubusercontent.com/4131505/93652826-459fcf80-f9cb-11ea-8bb2-feee99cd7cf6.png)

Tiny YOLOv4 model's cfg:
[tiny-yolov4.txt](https://github.com/AlexeyAB/darknet/files/5248636/tiny-yolov4.txt)

Full sized YOLOv4 model's cfg:
[yolov4.txt](https://github.com/AlexeyAB/darknet/files/5248637/yolov4.txt)

Any insight as to why the full sized YOLOv4 model is underperforming so severely on the same dataset?!

Additionally, we have a separate dataset (still just people, but from an inline perspective instead of overhead), and the tiny model also vastly out performs the full sized model.

**Comment by agjunyent** [comment]
I think that one explanation could be that the data is 'easy' to learn, and yolov4-tiny is able to learn it while yolov4 is too complex for such a simple dataset.

A larger model is not always better if data is easy

**Comment by a-rich** [comment]
@agjunyent I took your suggestion and tried training with the same data and same config (except for different anchors and input shape) and I'm getting poor validation accuracy still...I had expected increasing input shape alone to increase accuracy a little bit. Is it possible that increasing input shape helps only up to a certain point before it actually decreases accuracy?

Our best model to date is a yolov4 tiny architecture with 384x384 input shape which we quantize, compile for edgetpu, and deploy to production -- what we're trying to achieve is training a better (even if just a minute amount) model which will not be quantized and will run as an auditing tool to detect when our production model is not performing as well...

Attached are the training charts and cfgs for the prod model and the one that we're attempting to use for audits.

![tiny_384_chart](https://user-images.githubusercontent.com/4131505/97342337-569df500-1843-11eb-971c-ae0df28563a7.png)

![tiny_1280_chart](https://user-images.githubusercontent.com/4131505/97342370-60275d00-1843-11eb-8633-bf43abdfceb4.png)

```
diff tiny_384_cfg.txt tiny_1280_cfg.txt
3,5c3,5
< subdivisions=2
< width=384
< height=384
---
> subdivisions=64
> width=1280
> height=1280
214c214
< anchors =  34, 44,  50, 84,  86, 82,  77,137, 153,108, 121,202
---
> anchors = 112,146, 173,262, 333,285, 229,426, 343,564, 491,667
262c262
< anchors =  34, 44,  50, 84,  86, 82,  77,137, 153,108, 121,202
---
> anchors = 112,146, 173,262, 333,285, 229,426, 343,564, 491,667
```

[tiny_384_cfg.txt](https://github.com/AlexeyAB/darknet/files/5447163/tiny_384_cfg.txt)


[tiny_1280_cfg.txt](https://github.com/AlexeyAB/darknet/files/5447164/tiny_1280_cfg.txt)
